python/tk_multi_publish2/dialog.py

```diff
@@ -900,6 +937,36 @@ def do_publish(self):
                 finally:
                     self._progress_handler.pop()
 
+            # ---- post publish (all items)
+
+            # we will run this even if the publish failed, but halt if
+            # processing was manually stopped
+            if not self._stop_processing_flagged:
+
+                self._progress_handler.set_phase(self._progress_handler.PHASE_POST_PUBLISH)
```

Comment:
why is everything else plugin based but these two hooks global?

I thought we were trying to promote more modular workflows with the publish2 and this design feels like we are back to 'global' thinking! Maybe we could discuss? What are the use cases? Could a publish plugin implement a cleanup method? Could there be a both a plugin cleanup and a global cleanup?

I think if we can advocate modular workflows through our designs, this will help our clients build robust pipelines. at the same time, we don't want to be a blockage, hindering them to build solutions.